backend-flask/tests/services/test_openai_integration.py

```
@@ -1,61 +1,76 @@
-    mock_generator.side_effect = Exception("API Error")
-
-    result = await generate_vocabulary(generator=mock_generator)
+    mock_client = MagicMock()
+    mock_client.chat.completions.create.side_effect = Exception("API Error")
 
-    assert "error" in result
-    assert "API Error" in result["error"]
+    with patch(
+        "app.services.openai_service.get_openai_client", return_value=mock_client
+    ):
+        with pytest.raises(Exception):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)